### DISCUSSION TITLE
budgets and cron

### DISCUSSION THREAD

**Discussion by nick2k3** [firefly-iii/firefly-iii#8801] [Firefly III]
Hi,
I have a question related to budgets: do I always need a cron to be setup in order to set the next month/week/year budgets?
Or only for particular scenarios?

**Comment by JC5** [comment]
You can always set them manually, if that's what you mean?

**Comment by nick2k3** [comment]
Indeed, I meant: if I don't set them manually on the next 1st May, AND if I don't setup a cron job, Firefly will keep the default 0 € values for all budgets. Right?

**Reply by JC5** [reply]
Exactly right.

**Reply by nick2k3** [reply]
thanks! so from [this](https://docs.firefly-iii.org/how-to/firefly-iii/advanced/cron/) page I understand the different techniques to call a cron job against firefly (I am currently running it through docker).

what I am missing is the "link" between the commands and the creation of the budgets. 
Also the 0 3 * * * corresponds to every day at 3 AM, should I edit this to ensure it runs at the turning of the month?

```
# cron job for Firefly III using Docker itself
0 3 * * * docker exec $(docker container ls -a -f name=firefly --format="{{.ID}}") /usr/local/bin/php /var/www/html/artisan firefly-iii:cron
```

**Reply by JC5** [reply]
Ah yes, you're right. As long as the cron runs on the first of the month, monthly budgets will be created. Keep in mind, weekly budgets will be skipped. Also, other features of the cron job may not work as expected (recurring transactions, for example).
